### DISCUSSION TITLE
Unable to stream UniFi G5 PTZ

### DISCUSSION THREAD

**Discussion by GizmoGoody** [koush/scrypted#1552] [Q&A]
I have a UniFi G3 Instant and G4 Doorbell Pro which work great through Scrypted to HomeKit.

Yesterday I installed the new G5 PTZ. It's working great in the UniFi Protect app and web ui, but I cannot get the RTSP(S) stream to work with Scrypted.

I've reviewed and configured the camera IAW the [codec settings](https://github.com/koush/scrypted/wiki/Codec-Settings) and even reduced the resolution and bitrate to 2K and 1.5Mbps in an attempt to reduce potential causes.

<img width="274" alt="image" src="https://github.com/user-attachments/assets/6fac2a86-ee31-4ae0-bbf3-49d18c3c689b">
<img width="271" alt="image" src="https://github.com/user-attachments/assets/7fe32306-311a-40b6-9130-ee05215d277b">

Here is a copy of the Scrypted log right after attempting to start the stream in Scrypted:

```
[WebRTC Plugin]: signalingState have-remote-offer
[WebRTC Plugin]: signalingState stable
[WebRTC Plugin]: iceGatheringState gathering
[WebRTC Plugin]: iceConnectionState completed
[WebRTC Plugin]: iceGatheringState complete
[WebRTC Plugin]: connectionState connecting
[WebRTC Plugin]: iceConnectionState checking
[WebRTC Plugin]: waiting ice connected
[WebRTC Plugin]: iceConnectionState connected
[WebRTC Plugin]: connectionState connected
[WebRTC Plugin]: Connection is local network: true 192.168.0.30 {
[WebRTC Plugin]:   ipv4: true,
[WebRTC Plugin]:   type: 'prflx',
[WebRTC Plugin]:   isLocalNetwork: true,
[WebRTC Plugin]:   destinationId: '192.168.0.30'
[WebRTC Plugin]: }
[WebRTC Plugin]: done waiting ice connected
[WebRTC Plugin]: Connection is local network: true 192.168.0.30 {
[WebRTC Plugin]:   ipv4: true,
[WebRTC Plugin]:   type: 'prflx',
[WebRTC Plugin]:   isLocalNetwork: true,
[WebRTC Plugin]:   destinationId: '192.168.0.30'
[WebRTC Plugin]: }
[RTC Connection Bridge]: Client Stream Profile {
[RTC Connection Bridge]:   transcodeBaseline: false,
[RTC Connection Bridge]:   sessionSupportsH264High: true,
[RTC Connection Bridge]:   maximumCompatibilityMode: false,
[RTC Connection Bridge]:   userAgent: 'Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/127.0.0.0 Safari/537.36 Edg/127.0.0.0',
[RTC Connection Bridge]:   capabilities: {
[RTC Connection Bridge]:     audio: { codecs: [Array], headerExtensions: [Array] },
[RTC Connection Bridge]:     video: { codecs: [Array], headerExtensions: [Array] }
[RTC Connection Bridge]:   },
[RTC Connection Bridge]:   screen: { devicePixelRatio: 1.5, width: 1707, height: 1067 }
[RTC Connection Bridge]: }
[RTC Connection Bridge]: a/v h264 undefined found h264 opus
[RTC Connection Bridge]: first video packet 865
[RTC Connection Bridge]: fua packet missing. skipping refragmentation. 5
[RTC Connection Bridge]: fua packet missing. skipping refragmentation. 5
[RTC Connection Bridge]: fua packet missing. skipping refragmentation. 5
[RTC Connection Bridge]: fua packet missing. skipping refragmentation. 5
[RTC Connection Bridge]: fua packet missing. skipping refragmentation. 5
[RTC Connection Bridge]: fua packet missing. skipping refragmentation. 5
[RTC Connection Bridge]: fua packet missing. skipping refragmentation. 5
[RTC Connection Bridge]: fua packet missing. skipping refragmentation. 5
[RTC Connection Bridge]: fua packet missing. skipping refragmentation. 5
[RTC Connection Bridge]: fua packet missing. skipping refragmentation. 5
[RTC Connection Bridge]: fua packet missing. skipping refragmentation. 5
[RTC Connection Bridge]: jitter buffer purged packet: 6030
[RTC Connection Bridge]: jitter buffer purged packet: 6031
[RTC Connection Bridge]: jitter buffer purged packet: 6032
[RTC Connection Bridge]: fua packet missing. skipping refragmentation. 5
[RTC Connection Bridge]: fua packet missing. skipping refragmentation. 5
[RTC Connection Bridge]: fua packet missing. skipping refragmentation. 5
[RTC Connection Bridge]: fua packet missing. skipping refragmentation. 5
[RTC Connection Bridge]: fua packet missing. skipping refragmentation. 5
[RTC Connection Bridge]: fua packet missing. skipping refragmentation. 5
```

The only line that seems to be relevant is `a/v h264 undefined found h264 opus`, but I'm not sure how to affect the h264 type other than disabling `Enhanced Encoding` which I've already done. Also, I've already tried to disable `Record AI Events`, but (1) that doesn't seem to help and (2) it's enabled on the other UniFi cameras which stream without issue. I've also tried to restart the G5 PTZ and the system I'm running Scrypted on.

Is the G5 PTZ supported yet and any pointers on what to troubleshoot next?

**Comment by koush** [comment]
it is supported. probably issue with protect.

**Reply by GizmoGoody** [reply]
Thanks. What/who should I report it to?

**Comment by GizmoGoody** [comment]
@koush  I reported this issue to Ubiquiti and pasted the exchange below. They seem to think something else is misconfigured. What are your thoughts and should I ask them anything else? For my other two UniFi camera, I don't think I did anything extra to get them into Scrypted.

----------
UI Support (Ubiquiti Help Center) 
Sep 4, 2024, 07:00 MDT 
Hello,
 
The Autogenerated link is an RTSPS URL and not RTSP, and that's why modification is needed.
 
**It must be due to some other misconfiguration that we cant judge with this method.**
 
If you are still having trouble, or if you have any additional questions or concerns, feel free to reply to this message directly.
Best,
UI Support
Ubiquiti Inc.

----------
OK, that did work, so thank you. But two questions remain:
 
1.	Why is this necessary…shouldn’t the autogenerated links just be formatted properly?
2.	Why doesn’t Scrypted work with my two other Protect cameras, but not with this camera?
 
Thank you,
<me>

----------
UI Support (Ubiquiti Help Center) 
Sep 4, 2024, 05:54 MDT 
Hello,
 
Our console has a feature called RTSP, and you can use RTSP streaming to see the live feed of the cameras, but for that, the VLC player is needed because the RTSP link can work on the VLC player.
 
The link will be generated from the UniFi Protect application >> UniFi Devices >> Settings >> Advanced. As I won't be able to provide you the correct link because the link is auto-generated, you just need to make some changes to that link and follow some steps on the VLC media player.

<screenshot of where to get the RTSPS link...which I already knew>

Please copy the link from there and make these changes only, as I'm mentioning below.
 
1) Suppose you'll get a link like this rtsps://192.168.11.168:7441/CvW63DlSp2VVng8g?enableSrt. Then rtsps--> rtsp, 7441-->7447 and remove the whole part from? sign(including ? sign)
 
The link should look like this - rtsp://192.168.11.168:7447/CvW63DlSp2VVng8g  (rtsps://192.168.11.168:7441/CvW63DlSp2VVng8g?enableSrt)
 
2)Then try this link to VLC media player by making your computer on the same network as that of the console and then you will be able to play the Real Time Streaming.

Best,
UI Support
Ubiquiti Inc.
